File: src/lib/scala/Slate.Common/src/main/scala/slate/common/query/ConditionGroup.scala
package slate.common.query


class ConditionGroup(val left:Any, val operator:String, val right:Any ) extends ICondition{

  override def toStringQuery:String =
  {
    getString(left) + " " + operator + " " + getString(right)
  }


  private def getString(item:Any):String =
  {
    item match {
      case null         => ""
      case s:ICondition => s.toStringQuery()
      case _            => item.toString
    }
  }
}

File: src/lib/kotlin/slatekit-query/src/main/kotlin/slatekit/query/ConditionGroup.kt
/**
 * <slate_header>
 * url: www.slatekit.com
 * git: www.github.com/code-helix/slatekit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * about: A tool-kit, utility library and server-backend
 * mantra: Simplicity above all else
 * </slate_header>
 */

package slatekit.query

data class ConditionGroup(val left: Any, val operator: String, val right: Any) : ICondition {

    override fun toStringQuery(): String = getString(left) + " " + operator + " " + getString(right)

    private fun getString(item: Any?): String =
            if (item == null) {
                ""
            } else (item as? ICondition)?.toStringQuery() ?: item.toString()
}
